README.md
# icon_bottomsheet

This is a customized class for bottomsheet with dynamic icon(widget) on top with various shapes. It uses the default bottomsheet imported from `package:flutter/material.dart`. The IconBottomSheet class then creates a tranaparent border for the iconWidget. The size of the padding is defined by iconPadding. The iconSize is the size of the iconWidget and iconShape is the shape of the iconWidget. There are various shapes for iconWidget as `circle`, `square`, `circleTop`, `circleBottom`. Below are the images for the various iconShapes of IconBottomSheet.


<img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/bf_c.png?raw=true" height="300"/> <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/bs_s.png?raw=true" height="300"/>  <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/bf_cb.png?raw=true" height="300"/>   <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/bs_ct.png?raw=true" height="300"/>

<img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/b_pc.png?raw=true" height="300"/> <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/bt_pc.png?raw=true" height="300"/>  <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/b_pl.png?raw=true" height="300"/>   <img src="https://github.com/amitmishra7/icon_bottomsheet/blob/main/assets/b_pr.png?raw=true" height="300"/>


# Arguments

| Argument           | Description                                                                                       
|--------------------|---------------------------------------------------------------------------------------------------
| child              | This is the child widget that will be displayed in the bottomsheet.                               
| iconWidget         | Widget displayed on top of bottomsheet.                                                
| iconSize           | Height/Width of the iconWidget. Defaults to 50 (height and width are same).
| iconPadding        | Transparent border padding around the iconWidget. Defaults to 16. Use 0 when not requried.
| position           | Helps to position your iconWidget on left, right or center. Defaults to center.
| iconBackgroundColor           | Defines the background color of your iconWidget. Defaults to white. In order to show no color use Colors.transparent.
| childBackgroundColor           | Defines the background color of your child widget. Defaults to white. 

# Position

| Argument               | Description                                                                                       
|------------------------|---------------------------------------------------------------------------------------------------
| Position.center       | Aligns the iconWidget to center of the bottomsheet. It is set by default if no position is passed as a param.                                
| Position.left    | Aligns the iconWidget to left of the bottomsheet. 
| Position.right | Aligns the iconWidget to right of the bottomsheet. 

# IconShape

| Argument               | Description                                                                                       
|------------------------|---------------------------------------------------------------------------------------------------
| IconShape.circle       | Creates circle shape background of iconWidget.                               
| IconShape.circleTop    | Creates circle shape from top and square shape from bottom for background of iconWidget.
| IconShape.circleBottom | Creates circle shape from bottom and square shape from top for background of iconWidget.
| IconShape.square       | Creates square shape background of iconWidget.


# Getting started

Add the package `icon_bottomsheet` to your pubspec.yaml file and do pub get.


# Usage

Firstly you need to change the canvas color in the theme of your material app so that you bottomsheet has transparent background. 

```dart
MaterialApp(
    title: 'Flutter Demo',
    theme: ThemeData(
    canvasColor: Colors.transparent,
    ),
    home: const MyHomePage(title: 'Icon BottomSheet'),
);
```

After that import showModalBottomSheet from `material.dart` and then import the `IconBottomSheet` from `package:icon_bottomsheet/icon_bottomsheet.dart` and pass the arguments as described above.

```dart
showModalBottomSheet<int>(
    context: context,
    builder: (context) {
    return const IconBottomSheet(
        iconWidget: Icon(
        Icons.cancel_outlined,
        color: Colors.white,
        size: 100,
        ),
        iconShape: IconShape.square,
        position: Position.right,
        iconBackgroundColor: Colors.redAccent,
        //childBackgroundColor: Colors.grey,
        iconSize: 120,
        iconPadding: 20,
        child: PaymentFailed(),
    );
    },
);
```

## Contributor
[Amit Mishra](https://github.com/amitmishra7) 

## Contribute 

If you have any ideas for extending this package or have found a bug, please contribute!

1. You'll need a GitHub account.
2. Fork the icon_bottomsheet repository.
3. Work on your feature/bug.
4. Create a pull request.
5. Star this project. ⭐
6. Become a hero!! 🎉


CHANGELOG.md
## 0.0.3

[0.0.2] - 07.02.2023

Added iconBackgroundColor, childBackgroundColor and position for iconWidget.

## 0.0.2

[0.0.2] - 06.02.2023

Updated documentation and added new images

## 0.0.1

[0.0.1] - 06.02.2023

Initial publish of project containing IconBottomSheet Demo app using icon_bottomsheet package

example/lib/kid.dart
import 'package:flutter/material.dart';

class Kid extends StatefulWidget {
  const Kid({Key? key}) : super(key: key);

  @override
  State<Kid> createState() => _KidState();
}

class _KidState extends State<Kid> {
  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: MediaQuery.of(context).size.width,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(
              height: 10,
            ),
            const Center(
              child: Text(
                'Hello Steve.!',
                style: TextStyle(
                    color: Colors.pinkAccent,
                    fontSize: 40,
                    fontWeight: FontWeight.w600),
              ),
            ),
            const SizedBox(
              height: 25,
            ),
            const Text(
              'Your request has been submitted. We will get back to you shortly. Meantime you can explore our products',
              style: TextStyle(
                  color: Colors.black87,
                  fontSize: 22,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 25,
            ),
            const SizedBox(
              height: 20,
            ),
            ClipRRect(
                borderRadius: BorderRadius.circular(10),
                child: Container(
                  color: Colors.yellow,
                  height: 50,
                  child: const Center(
                      child: Text(
                    'Explore More',
                    style: TextStyle(fontSize: 18),
                  )),
                ))
          ],
        ),
      ),
    );
  }
}


example/lib/success.dart
import 'package:flutter/material.dart';

class Success extends StatefulWidget {
  const Success({Key? key}) : super(key: key);

  @override
  State<Success> createState() => _SuccessState();
}

class _SuccessState extends State<Success> {
  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: MediaQuery.of(context).size.width,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(
              height: 10,
            ),
            const Center(
              child: Text(
                'Failure',
                style: TextStyle(
                    color: Colors.redAccent,
                    fontSize: 40,
                    fontWeight: FontWeight.w600),
              ),
            ),
            const SizedBox(
              height: 25,
            ),
            const Text(
              'Your application has been rejected. Please fill the form properly and upload again.',
              style: TextStyle(
                  color: Colors.black87,
                  fontSize: 22,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 25,
            ),
            const SizedBox(
              height: 20,
            ),
            ClipRRect(
                borderRadius: BorderRadius.circular(10),
                child: Container(
                  color: Colors.yellow,
                  height: 50,
                  child: const Center(
                      child: Text(
                    'OKAY',
                    style: TextStyle(fontSize: 18, fontWeight: FontWeight.w600),
                  )),
                ))
          ],
        ),
      ),
    );
  }
}


lib/icon_bottomsheet.dart
import 'package:flutter/material.dart';

/// IconBottomSheet is a simple class developed for customized bottomSheet. This class creates a circular/square shape icon on the top of bottomSheee. It creates a circular icon with transparent padding on top of the bottomSheet. This circular widget is half overlapped on the centerTop of the bottomSheet. The bottomSheet is circular from the topLeft and topRight.

enum IconShape { circle, square, circleTop, circleBottom }

enum Position { center, left, right }

class IconBottomSheet extends StatelessWidget {
  const IconBottomSheet({
    Key? key,
    required this.child,
    required this.iconWidget,
    this.iconSize = 50,
    this.iconPadding = 16,
    this.iconShape = IconShape.circle,
    this.iconBackgroundColor = Colors.white,
    this.childBackgroundColor = Colors.white,
    this.position = Position.center,
  }) : super(key: key);

  /// [optional] iconSize resembles the height (same as width) of the icon widget that will be displayed.
  /// The iconSize should always be same as the height(same as width) of the iconWidget. Defaults to 50.
  final double iconSize;

  /// [optional] iconPadding resembles the padding from all sides to the iconWidget.
  /// iconPadding is transparent border around the iconWidget. Defaults to 16.
  final double iconPadding;

  /// [mandatory] child is the view that we are displaying in the bottomSheet.
  /// The child will always be displayed below the iconWidget.
  final Widget child;

  /// [mandatory] iconWidget is the widget that is displayed on top of bottomSheet.
  /// The height (same as width) of the iconWidget should always be same as the iconSize.
  final Widget iconWidget;

  /// [optional] iconShape defines the shape of the iconWidget. Defaults to circular
  final IconShape iconShape;

  /// [optional] iconBackground color is the color of the iconWidget background. Defaults to white.
  final Color iconBackgroundColor;

  /// [optional] childBackgroundColor color is the color of the child background. Defaults to white.
  final Color childBackgroundColor;

  /// [optional] to position the iconWidget. Defaults to center.
  final Position position;

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Stack(
        children: [
          Padding(
            padding: EdgeInsets.only(top: iconSize / 2),
            child: ClipRRect(
              borderRadius: const BorderRadius.only(
                  topLeft: Radius.circular(15), topRight: Radius.circular(15)),
              child: ColorFiltered(
                colorFilter:
                    ColorFilter.mode(childBackgroundColor, BlendMode.srcOut),
                child: Stack(
                  children: [
                    Container(
                      decoration: const BoxDecoration(
                          color: Colors.white,
                          backgroundBlendMode: BlendMode.dstOut),
                    ),
                    Positioned(
                      top: -iconSize / 2,
                      left: iconWidgetOuterPosition(context),
                      //(MediaQuery.of(context).size.width - iconSize - iconPadding) / 2,
                      child: Container(
                        height: iconSize + iconPadding,
                        width: iconSize + iconPadding,
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: borderRadius,
                        ),
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ),
          Positioned(
            top: iconPadding / 2,
            left: iconWidgetInnerPosition(context),
            child: Container(
              height: iconSize,
              width: iconSize,
              decoration: BoxDecoration(
                color: iconBackgroundColor, //Colors.white,
                borderRadius: borderRadius,
              ),
              child: iconWidget,
            ),
          ),
          Positioned(top: iconSize + iconPadding, child: child)
        ],
      ),
    );
  }

  BorderRadius? get borderRadius {
    switch (iconShape) {
      case IconShape.circle:
        return BorderRadius.circular(500);
      case IconShape.square:
        return null;
      case IconShape.circleTop:
        return const BorderRadius.only(
            topRight: Radius.circular(500), topLeft: Radius.circular(500));
      case IconShape.circleBottom:
        return const BorderRadius.only(
            bottomRight: Radius.circular(500),
            bottomLeft: Radius.circular(500));
      default:
        return BorderRadius.circular(500);
    }
  }

  double iconWidgetOuterPosition(context) {
    switch (position) {
      case Position.left:
        return 0;
      case Position.right:
        return (MediaQuery.of(context).size.width - iconSize - iconPadding);
      case Position.center:
        return (MediaQuery.of(context).size.width - iconSize - iconPadding) / 2;
    }
  }

  double iconWidgetInnerPosition(context) {
    switch (position) {
      case Position.left:
        return iconPadding / 2;
      case Position.right:
        return (MediaQuery.of(context).size.width - iconSize - iconPadding / 2);
      case Position.center:
        return (MediaQuery.of(context).size.width - iconSize) / 2;
    }
  }
}


lib/icon_bottom_sheet.dart
import 'package:flutter/material.dart';

/// IconBottomSheet is a simple class developed for customized bottomSheet. This class creates a circular/square shape icon on the top of bottomSheee. It creates a circular icon with transparent padding on top of the bottomSheet. This circular widget is half overlapped on the centerTop of the bottomSheet. The bottomSheet is circular from the topLeft and topRight.

enum IconShape { circle, square, circleTop, circleBottom }

class IconBottomSheet extends StatelessWidget {
  const IconBottomSheet({
    Key? key,
    required this.child,
    required this.iconWidget,
    this.iconSize = 50,
    this.iconPadding = 16,
    this.iconShape = IconShape.circle,
  }) : super(key: key);

  /// [optional] iconSize resembles the height (same as width) of the icon widget that will be displayed.
  /// The iconSize should always be same as the height(same as width) of the iconWidget. Defaults to 50.
  final double iconSize;

  /// [optional] iconPadding resembles the padding from all sides to the iconWidget.
  /// iconPadding is transparent border around the iconWidget. Defaults to 16.
  final double iconPadding;

  /// [mandatory] child is the view that we are displaying in the bottomSheet.
  /// The child will always be displayed below the iconWidget.
  final Widget child;

  /// [mandatory] iconWidget is the widget that is displayed on top of bottomSheet.
  /// The height (same as width) of the iconWidget should always be same as the iconSize.
  final Widget iconWidget;

  /// [optional] iconShape defines the shape of the iconWidget. Defaults to circular
  final IconShape iconShape;

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Stack(
        children: [
          Padding(
            padding: EdgeInsets.only(top: iconSize / 2),
            child: ClipRRect(
              borderRadius: const BorderRadius.only(topLeft: Radius.circular(15), topRight: Radius.circular(15)),
              child: ColorFiltered(
                colorFilter: const ColorFilter.mode(Colors.white, BlendMode.srcOut),
                child: Stack(
                  children: [
                    Container(
                      decoration: const BoxDecoration(color: Colors.white, backgroundBlendMode: BlendMode.dstOut),
                    ),
                    Positioned(
                      top: -iconSize / 2,
                      left: (MediaQuery.of(context).size.width - iconSize - iconPadding) / 2,
                      child: Container(
                        height: iconSize + iconPadding,
                        width: iconSize + iconPadding,
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: borderRadius,
                        ),
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ),
          Positioned(
            top: iconPadding / 2,
            left: (MediaQuery.of(context).size.width - iconSize) / 2,
            child: Container(
              height: iconSize,
              width: iconSize,
              decoration: BoxDecoration(
                color: Colors.white,
                borderRadius: borderRadius,
              ),
              child: iconWidget,
            ),
          ),
          Positioned(top: iconSize + iconPadding, child: child)
        ],
      ),
    );
  }

  BorderRadius? get borderRadius {
    switch (iconShape) {
      case IconShape.circle:
        return BorderRadius.circular(500);
      case IconShape.square:
        return null;
      case IconShape.circleTop:
        return const BorderRadius.only(topRight: Radius.circular(500), topLeft: Radius.circular(500));
      case IconShape.circleBottom:
        return const BorderRadius.only(bottomRight: Radius.circular(500), bottomLeft: Radius.circular(500));
      default:
        return BorderRadius.circular(500);
    }
  }
}


example/lib/payment_failed.dart
import 'package:flutter/material.dart';

class PaymentFailed extends StatefulWidget {
  const PaymentFailed({Key? key}) : super(key: key);

  @override
  State<PaymentFailed> createState() => _PaymentFailedState();
}

class _PaymentFailedState extends State<PaymentFailed> {
  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: MediaQuery.of(context).size.width,
      child: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(
              height: 10,
            ),
            const Center(
              child: Text(
                'Payment Failed.!',
                style: TextStyle(
                    color: Colors.redAccent,
                    fontSize: 30,
                    fontWeight: FontWeight.w600),
              ),
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            const Text(
              'Dont worry your money is safe! If money was debited from your account, it will be refunded automatically in 5-7 working days.',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: const [
                Text(
                  'Payment ID',
                  style: TextStyle(
                      color: Colors.black87,
                      fontSize: 16,
                      fontWeight: FontWeight.w600),
                ),
                Text(
                  'PP782365479193984749',
                  style: TextStyle(
                      color: Colors.black54,
                      fontSize: 16,
                      fontWeight: FontWeight.w400),
                ),
              ],
            ),
            const SizedBox(
              height: 10,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: const [
                Text(
                  'Time',
                  style: TextStyle(
                      color: Colors.black87,
                      fontSize: 16,
                      fontWeight: FontWeight.w600),
                ),
                Text(
                  '14th June 2022 at 8:45pm',
                  style: TextStyle(
                      color: Colors.black54,
                      fontSize: 16,
                      fontWeight: FontWeight.w400),
                ),
              ],
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            const Text(
              'Why did it fail?',
              style: TextStyle(
                  color: Colors.black87,
                  fontSize: 16,
                  fontWeight: FontWeight.w600),
            ),
            const SizedBox(
              height: 5,
            ),
            const Text(
              '1. The CVV or Expiry Date might be wrong.',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const Text(
              '2. Possibly your bank server might be down',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 20,
            ),
            ClipRRect(
                borderRadius: BorderRadius.circular(10),
                child: Container(
                  color: Colors.green,
                  width: double.infinity,
                  height: 40,
                  child: const Center(child: Text('Retry')),
                ))
          ],
        ),
      ),
    );
  }
}


example/lib/payment_success.dart
import 'package:flutter/material.dart';

class PaymentSuccess extends StatefulWidget {
  const PaymentSuccess({Key? key}) : super(key: key);

  @override
  State<PaymentSuccess> createState() => _PaymentSuccessState();
}

class _PaymentSuccessState extends State<PaymentSuccess> {
  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: MediaQuery.of(context).size.width,
      child: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(
              height: 10,
            ),
            const Center(
              child: Text(
                'Payment Successful.!',
                style: TextStyle(
                    color: Colors.green,
                    fontSize: 30,
                    fontWeight: FontWeight.w600),
              ),
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            const Text(
              'Thank you for your payment. Your order has been placed and will be delivered shortly. We will be in contact with more details shortly',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: const [
                Text(
                  'Payment ID',
                  style: TextStyle(
                      color: Colors.black87,
                      fontSize: 16,
                      fontWeight: FontWeight.w600),
                ),
                Text(
                  'PP782365479193984749',
                  style: TextStyle(
                      color: Colors.black54,
                      fontSize: 16,
                      fontWeight: FontWeight.w400),
                ),
              ],
            ),
            const SizedBox(
              height: 10,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: const [
                Text(
                  'Time',
                  style: TextStyle(
                      color: Colors.black87,
                      fontSize: 16,
                      fontWeight: FontWeight.w600),
                ),
                Text(
                  '14th June 2022 at 8:45pm',
                  style: TextStyle(
                      color: Colors.black54,
                      fontSize: 16,
                      fontWeight: FontWeight.w400),
                ),
              ],
            ),
            const SizedBox(
              height: 10,
            ),
            const Divider(
              height: 2,
              color: Colors.grey,
            ),
            const SizedBox(
              height: 10,
            ),
            const Text(
              'Track your order',
              style: TextStyle(
                  color: Colors.black87,
                  fontSize: 16,
                  fontWeight: FontWeight.w600),
            ),
            const SizedBox(
              height: 5,
            ),
            const Text(
              '1. Go to My Orders Section',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const Text(
              '2. Click on a particular order to see more details',
              style: TextStyle(
                  color: Colors.black54,
                  fontSize: 16,
                  fontWeight: FontWeight.w400),
            ),
            const SizedBox(
              height: 20,
            ),
            ClipRRect(
                borderRadius: BorderRadius.circular(10),
                child: Container(
                  color: Colors.green,
                  width: double.infinity,
                  height: 40,
                  child: const Center(child: Text('Download Receipt')),
                ))
          ],
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:example/kid.dart';
import 'package:example/payment_failed.dart';
import 'package:example/payment_success.dart';
import 'package:example/success.dart';
import 'package:flutter/material.dart';
import 'package:icon_bottomsheet/icon_bottomsheet.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
        canvasColor: Colors.transparent,
      ),
      home: const MyHomePage(title: 'Icon BottomSheet'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  void _showBottomSheet() {
    showModalBottomSheet<int>(
      context: context,
      builder: (context) {
        return const IconBottomSheet(
          iconWidget: Icon(
            Icons.cancel_outlined,
            color: Colors.redAccent,
            size: 100,
          ),
          position: Position.right,
          iconShape: IconShape.circle,
          iconSize: 120,
          iconPadding: 20,
          child: Success(),
        );
      },
    );
  }

  void _showKidsBottomSheet() {
    showModalBottomSheet<int>(
      context: context,
      builder: (context) {
        return IconBottomSheet(
          iconWidget: Padding(
            padding: const EdgeInsets.all(16.0),
            child: Image.asset('assets/duckling.png'),
          ),
          //iconBackgroundColor: Colors.transparent,
          //childBackgroundColor: Colors.blue,
          iconShape: IconShape.circle,
          iconSize: 120,
          iconPadding: 20,
          child: const Kid(),
        );
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            const SizedBox(
              height: 20,
            ),
            ElevatedButton(
              onPressed: _showBottomSheet,
              child: const Text(
                'Show BottomSheet',
              ),
            )
          ],
        ),
      ),
    );
  }
}
